### PR TITLE
Fix Form should only trigger field validation on submit if not told otherwise

### PR DESCRIPTION
**Pull request by slax57**
Fix https://github.com/marmelab/react-admin/issues/8799

## Todo

- [x] Fix
- [x] Test
- [x] Non-regression tests with EE components
- [x] Mention the related RHF rule in OSS docs

### CODE REVIEW

File: docs/Inputs.md [hunk 1]
```diff
@@ -891,6 +891,13 @@ const PersonEdit = () => (
 );
 ```
 
+**Reminder:** [react-hook-form's `formState` is wrapped with a Proxy](https://react-hook-form.com/api/useformstate/#rules) to improve render performance and skip extra computation if specific state is not subscribed. So, make sure you deconstruct or read the `formState` before render in order to enable the subscription.
```
Comment (djhi) [inline docs/Inputs.md:894]: This one is weird as we don't use `useFormState` anywhere before or did I miss something?

File: docs/useInput.md [hunk 2]
```diff
@@ -100,6 +100,13 @@ const LatLngInput = props => {
 };
 ```
 
+**Reminder:** [react-hook-form's `formState` is wrapped with a Proxy](https://react-hook-form.com/api/useformstate/#rules) to improve render performance and skip extra computation if specific state is not subscribed. So, make sure you deconstruct or read the `formState` before render in order to enable the subscription.
```
Comment (djhi) [inline docs/useInput.md:103]: Same

File: docs/useInput.md [hunk 3]
```diff
@@ -146,3 +153,10 @@ const PersonEdit = () => (
     </Edit>
 );
 ```
+
```
Comment (djhi) [inline docs/useInput.md:156]: Same

**Review by djhi — CHANGES_REQUESTED** [review]

File: docs/Inputs.md (hunk 1 above)
Comment (slax57) [reply docs/Inputs.md:894]: We use the `formState` (`formState: { isSubmitted }`) so I figured I'd rather add the Reminder here anyway. But I'm open to discussion :grin:

**Review by djhi — APPROVED** [review]

**Pull request merged**